src/network/networktool.cpp

```diff
@@ -0,0 +1,178 @@
+#include "logger.h"
+#include "maddy/parser.h"
+#include "networkmanager.h"
+#include "networktool.h"
+#include "notificationmanager.h"
+
+#include <QJsonArray>
+#include <QJsonDocument>
+#include <QRegularExpression>
+#include <QTimer>
+#include <QUrl>
+
+PING_LOGGING_CATEGORY(NETWORKTOOL, "ping.networktool")
+
+void NetworkTool::checkInterfaceUpdate()
+{
+    static QString gitUserRepo;
+    if(gitUserRepo.isEmpty()) {
+        //https://github.com/user/repo.git
+        const static QRegularExpression regex("github.com/(.*).git");
+        QRegularExpressionMatch regexMatch = regex.match(QStringLiteral(GIT_URL));
+        // TODO: Check regex output
+        if (regexMatch.hasMatch()) {
+            gitUserRepo = regexMatch.capturedTexts()[1];
+        } else {
+            qCWarning(NETWORKTOOL) <<
+                                   "Fail to get github user and repository! It'll not be possible to check for updates";
+            return;
+        }
+    }
+
+    const QUrl url{QStringLiteral("https://api.github.com/repos/%1/releases").arg(gitUserRepo)};
+    NetworkManager::self()->requestJson(
+        url,
+        self()->checkNewVersionInGitHubPayload
+    );
+}
+
+void NetworkTool::checkNewVersionInGitHubPayload(QJsonDocument& jsonDocument)
+{
+    const static QString projectTag = QStringLiteral(GIT_TAG);
+    static QRegularExpression releaseTagRegex;
+    // Check if version start with 't' from test version
+    // For everything we can presume normal release
+    if(projectTag.contains('t') == 0) {
+        releaseTagRegex.setPattern(QStringLiteral("(t)\\d+\\.\\d+"));
+        qCDebug(NETWORKTOOL) << "Running test version:" << projectTag;
+    } else {
+        releaseTagRegex.setPattern(QStringLiteral("(v)\\d+\\.\\d+"));
+        qCDebug(NETWORKTOOL) << "Running release version:" << projectTag;
+    }
+
+    // Check if this actual version is a real release
+    // Alarm people to download the correction version !
+    QRegularExpressionMatch regexMatch = releaseTagRegex.match(projectTag);
+    float actualVersion = regexMatch.hasMatch() ? projectTag.right(1).toFloat() : -1.0;
+
+    auto versionsAvailable = jsonDocument.array();
+    if(versionsAvailable.isEmpty()) {
+        qCWarning(NETWORKTOOL) << "GitHub json it's not an array!";
+        qCDebug(NETWORKTOOL) << jsonDocument;
+        return;
+    }
+
+    struct {
+        float version;
+        QString versionString;
+        QJsonValue json;
+    } lastReleaseAvailable;
+    lastReleaseAvailable.version = actualVersion;
+
+    // Check payload
+    for(const QJsonValue& versionPayload : versionsAvailable) {
+        auto versionString = versionPayload[QStringLiteral("tag_name")].toString();
+        qCDebug(NETWORKTOOL) << "Testing version:" << versionString;
+
+        regexMatch = releaseTagRegex.match(versionString);
```

Comment:
If we are currently running a test release, we will only be checking for test releases upstream
If we are currently running a regular release, we will only be checking for regular releases upstream (see above lines 46 and 49). Is this ok?

Reply:
Yes, that's the idea. Anyone that want to be bleeding edge or want to be a beta testerof ping-viewer will receive notifications about new test releases (a.k.a Beta).

Reply:
But these same people will not catch the stable release, even when it is newer than their beta/test.

Reply:
Done.